pkg/pillar/Makefile

```diff
@@ -76,6 +76,12 @@ build-docker-$(APPS): $(DISTDIR)
 build-docker:
 	docker build $(DOCKER_ARGS) -t $(DOCKER_TAG) .
 
+build-docker-dev:
+	docker build $(DOCKER_ARGS) -t $(DOCKER_TAG) . --target build
```

Comment:
@OhmSpectator , I liked these targets, but unfortunately, they don't consider architecture, as the whole Makefile. In a nutshell, I cannot run enter-docker-dev for ARM.... You don't need to add it in this PR, but it would be good to have architecture support in the future, maybe we could create an issue/ticket for that...

Reply:
Hmmm... Good point. And just a question of curiosity, how do you make ARCH-specific steps for building Pillar now? Is there anything build-time specific in general?

Reply:
It's handled inside the Dockerfile to setup corresponding GO variables and cross-compilers (when needed): https://github.com/lf-edge/eve/blob/master/pkg/pillar/Dockerfile#L86

Reply:
So, the problem is that TARGETARCH is not set from Makefile, yeah?

Reply:
yes for `go build ...` commands, but in case of docker commands, it should be platform, like: `--platform linux/arm64`

Reply:
I think this is the only remaining comment before this can be merged.

Reply:
Yeah. I tried to devise a quick fix, but it did not work as expected. I'll come up with another solution and push it here. Meanwhile, switching the PR into Draft.

Reply:
@eriknordmark, as the new targets are not the only ones that lack multi-arch support, I would prefer to address this issue in the scope of a dedicated task, not blocking the PR. I'll create a ticket for that shortly.